src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -65,7 +65,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                         //before the item). In these cases we will create a new object and read data from the JObject
                         dataReader = new ForkableJsonReader(existingValueJObject.CreateReader(), dataReader.SerializationDataToken);
                         dataReader.Read(); //JObject readers begin at Not Started
-                        existingValue = contract.DefaultCreator();
+                        existingValue = CreateDefault(contract, reference.Type);

```

Comment:
This might have issues because the `contract` object we have may no longer be applicable to the object we created.

It might be simpler to resolve the contract twice. Once in the CreateType  (passing parametesr  `CreateDefault(Type objectType, string jsonApiType, Serializer serializer)`)  method. Then again inside `PopulateProperties` (dont pass the contract just resolve it from the obj.GetType())